Project/src/main/java/app/ReviewerController.java

```diff
@@ -21,35 +33,53 @@
 
     /***
-     * Gets the information about a single article
-     * @param model The model to return the files
-     * @param id The id of the article
-     * @return The review view
+     * Uploads a file to a review
+     * @param file The Multipart file that got uploaded
+     * @return
+     * @throws IOException
      */
-    @GetMapping("/review/{id}")
-    public String getReviewer(Model model, @PathVariable String id) {
-        model.addAttribute("article", articleRepository.findById(Long.parseLong(id)).get());
-        return "review";
+    @PostMapping("/review")
+    public String uploadFile(@RequestParam String articleId, @RequestParam("file") MultipartFile file) throws IOException {
+        if (file == null || StringUtils.isNullOrEmpty(file.getOriginalFilename())
+                || StringUtils.isNullOrEmpty(file.getContentType()) || file.getBytes().length <= 0) {
+            //This is how spring handles bad requests, throws exceptions.
+            throw new BadFileException();
+        }
+        articleRepository.findById(Long.parseLong(articleId)).ifPresent((Article article)  -> {
+            try{
+                article.setReview(file.getBytes());
+                article.setReviewFileType(file.getContentType());
+            } catch(IOException e) {}
+            articleRepository.save(article);
+        });
+        return "redirect:/reviewer";
     }
 
     /***
-     * Sets the new review and redirects to /reviewer
-     * @param model The model to return the files
-     * @param articleId The id of the article
-     * @param review The review
-     * @return Redirect to the reviewer view
+     * Downloads a review
+     * @param id The ID of the file whose review is to be downloaded
+     * @return
      */
-    @PostMapping("/setReview")
-    public String setReview(@RequestParam String review, @RequestParam String articleId) {
-        articleRepository.findById(Long.parseLong(articleId)).ifPresent((Article article) -> {
-            article.setReview(review);
-            articleRepository.save(article);
-        });
-        return "redirect:/reviewer";
+    @GetMapping("/review/{id}")
+    public ResponseEntity<byte[]> getFileById(@PathVariable(value = "id")Long id){
+        Article article = articleRepository.findById(id).get();
+        User currentUser = super.getUser();
+        // allow editors to download all files
+        if (!article.getUsers().contains(currentUser) && currentUser.getRole() != Role.EDITOR) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.parseMediaType(article.getReviewFileType()));
+        //sets the file name
+        headers.setContentDispositionFormData("review-" + id + ".txt", "review-" + id + ".txt");
+        //I admittedly have no idea what these cache control things are
+        headers.setCacheControl("must-revalidate, post-check=0, pre-check=0");
```

Comment:
If you dont know what this line does why is it here

Reply:
That's from Bhavik not me